library/core/tests/char.rs

```diff
@@ -413,3 +413,58 @@ fn eu_iterator_specializations() {
     check('\u{12340}');
     check('\u{10FFFF}');
 }
+
+macro_rules! is_ascii_tests {
+    ($x: ident $(, $test_name: ident, $method_name: ident, $original_version: expr)+ $(,)?) => {
+        $(
+            #[test]
+            fn $test_name() {
+                for $x in ('\0'..='\u{d7ff}').chain('\u{e000}'..='\u{10ffff}') {
```

Comment:
You don't need to do this, just `'\0'..char::MAX` works. The invalid ranges will just be skipped automatically.

Reply:
Fixed in commit 8.